src/services/permanence-conseillers/permanence/repositories/permanence-conseiller.repository.js

```diff
@@ -1,27 +1,40 @@
 const { ObjectId } = require('mongodb');
 
-const getPermanences = db => async () =>
-  await db.collection('permanences').aggregate([
-    {
-      $lookup: {
-        localField: 'structure.$id',
-        from: 'structures',
-        foreignField: '_id',
-        pipeline: [
-          {
-            $project: {
-              _id: 0,
-              siret: 1,
-              estLabelliseFranceServices: 1,
-              estLabelliseAidantsConnect: 1
-            }
+const getPermanences = db => async () => await db.collection('permanences').aggregate([
+  {
+    $addFields: {
+      'entity': {
+        $arrayElemAt: [{ $objectToArray: '$structure' }, 1]
+      }
+    }
+  },
+  {
+    $addFields: {
+      'entity': '$entity.v'
+    }
+  },
+  {
+    $lookup: {
+      from: 'structures',
+      let: { idStructure: '$entity' },
```

Comment:
@squirrion, c'est possible d'en supprimer un des deux de cette manière :
```suggestion
  {
    $addFields: {
      'entity': {
        $arrayElemAt: [{ $objectToArray: '$structure' }, 1]
      }
    }
  },
  {
    $lookup: {
      from: 'structures',
      let: { idStructure: '$entity.v' },
```

Reply:
effectivement 👍